packages/server/test/integration/http_requests_spec.js

```diff
@@ -3783,6 +3783,60 @@ describe('Routes', () => {
       })
     })
 
+    // Set custom status message/reason phrase from http response
+    // https://github.com/cypress-io/cypress/issues/16973
+    it('set custom status message when reason phrase is set', function () {
+      nock(this.server.remoteStates.current().origin)
+      .post('/companies/validate', {
+        payload: { name: 'Brian' },
+      })
+      .reply(400, undefined, {
+        'Reason-Phrase': 'Custom Reason Phrase Message',
```

Comment:
Again, the `reason-phrase` header isn't what we're looking for. We're looking for the HTTP reason phrase: https://www.w3.org/Protocols/rfc2616/rfc2616-sec6.html

Apparently it's tricky to mock this with nock, but it's possible, here's how: https://github.com/nock/nock/issues/1979#issue-604723095

Reply:
Thank you @flotwig for taking out time to review and share your feedback in the PR. In my Organisation, we have our VP visiting us this week, so I will be pretty occupied. Will definitely try to take this up this weekend. 
For http reason phrase, I was also following the same URL - https://www.w3.org/Protocols/rfc2616/rfc2616-sec6.html
However, I have misinterpreted it that it is set in the HTTP headers. However, I realise that you are correct and would update the implementation as per your comment and will also combine the middleware.

Reply:
@flotwig - Updated the test cases as per your suggestion. Thank you for sharing mocking the implementation in nock. It made the implementation straight-forward.